tests/test_api.py

```diff
@@ -59,6 +59,29 @@ def test_local_with_config(self):
         assert cover.config['repo_token'] == 'xxx'
         assert 'service_job_id' not in cover.config
 
+    def generate_import_mock(self, bad_module_name, msg):
+        """ Returns replacement for builting import function, which raises 
+        exception on importing appropriate module"""
+
+        origin = __import__
+        def import_func(name, *args):
+            if name == bad_module_name:
+                raise ImportError(msg)
+            else:
+                return origin(name, *args)
+        return import_func
+
+    @patch.object(log, 'warn')
+    def test_local_with_config_without_yaml_module(self, mock_logger):
+        """test local with config in yaml, but without yaml-installed"""
+        
+        yaml_import_mock = self.generate_import_mock('yaml', 'No module named yaml')
+        try:
+            with patch('__builtin__.__import__', side_effect=yaml_import_mock):
+                Coveralls()
+        except:
```

Comment:
Why would you want to catch all exceptions, even those you're not expecting to get?